lib/ansible/playbook/conditional.py

```diff
@@ -176,7 +176,7 @@ def generic_visit(self, node, inside_call=False, inside_yield=False):
                         )
             try:
                 e = templar.environment.overlay()
-                e.filters.update(templar._get_filters())
+                e.filters.update(templar._get_filters(e.filters))
```

Comment:
could reverse the update ordering and update templar._get_filters() (or a copy...) with e.filters so that
the e.filters 'wins'

Reply:
something like

``` python
custom_filters = templar._get_filters()
custom_filters.update(e.filters)
e.filters = custom_filters
```

Reply:
We may want to override a builtin filter with an ansible provided by filter, such as `groupby`.  We just don't want this to happen with custom tests.

Eventually custom tests will no longer be registered as filters (2.9), but have been kept thus far for backwards compat.